### ISSUE TITLE
Logcollector does not reconnect to the agent daemon and floods ossec.log

### ISSUE THREAD

**Issue by Molter73** [wazuh/wazuh#4429]
|Wazuh version|Component|Install type|
|---|---|---|
| 3.12 | logcollector | Agent |

If agentd is killed by any abnormal situation, logcollector starts writing the following message into ossec.log: `2020/01/09 17:42:26 ossec-logcollector: ERROR: socketerr (not available).`
This message is written once for every time logcollector attempts to write to agentd, causing the size of ossec.log to grow fast.

Additionally, if the agentd daemon is restarted by executing ossec-agentd, logcollector does not attempt to reconnect to agentd and keeps writting the error:
```
2020/01/09 17:42:21 ossec-agentd: INFO: Using notify time: 10 and max time to reconnect: 60
2020/01/09 17:42:21 ossec-agentd: INFO: Version detected -> Linux |centos7 |3.10.0-957.12.2.el7.x86_64 |#1 SMP Tue May 14 21:24:32 UTC 2019 |x86_64 [CentOS Linux|centos: 7.6] - Wazuh v3.12.0
2020/01/09 17:42:21 ossec-agentd: INFO: (1410): Reading authentication keys file.
2020/01/09 17:42:21 ossec-agentd: INFO: Using AES as encryption method.
2020/01/09 17:42:21 ossec-agentd: INFO: Started (pid: 10457).
2020/01/09 17:42:21 ossec-agentd: INFO: Server IP Address: 192.168.50.10
2020/01/09 17:42:21 ossec-agentd: INFO: Trying to connect to server (192.168.50.10:1514/udp).
2020/01/09 17:42:21 ossec-logcollector: WARNING: Process locked due to agent is offline. Waiting for connection...
2020/01/09 17:42:22 ossec-agentd: INFO: (4102): Connected to the server (192.168.50.10:1514/udp).
2020/01/09 17:42:26 ossec-logcollector: INFO: Agent is now online. Process unlocked, continuing...
2020/01/09 17:42:26 ossec-logcollector: ERROR: socketerr (not available).
2020/01/09 17:42:26 ossec-logcollector: ERROR: socketerr (not available).
2020/01/09 17:42:26 ossec-logcollector: ERROR: socketerr (not available).
```

**Comment by Molter73** [comment]
**Proposed solution:**
- Prevent logcollector from printing the message out on the log everytime an event is triggered.
- Make logcollector attempt to reconnect to agentd so a full agent restart is not mandatory.
- Modify the log message so it displays a more meaningful description of the error.

**Comment by chemamartinez** [comment]
Merged #4435
